lib/topology_lib_tcpdump/library.py

```diff
@@ -49,39 +55,49 @@ def tcpdump_rate(sw):
     rate = int(total_packets) * 1000 / msec
     return rate
 
+"""
+Start packet capture using tcpdump.
+:param enode: Engine node to communicate with.
+:type enode: topology.platforms.base.BaseNode
+:param str interface_id: interface name.
+:param int wait_time: Time in seconds to capture with tcpdump.
+:param int num_cpu_samples: Number of CPU samples to get CPU utilization.
+ Default is 0
+"""
 
-def tcpdump_capture_interface(sw, options, interface_id, wait_time, check_cpu):
-    cmd_output = sw('ip netns exec swns tcpdump -D'.format(**locals()),
-                    'bash')
+def tcpdump_capture_interface(enode, options, interface_id, wait_time,
```

Comment:
We are currently the only consumers of this API. This signature makes it more generic.